spec/lib/open_project/plugins/module_hanler_spec.rb

```diff
@@ -0,0 +1,38 @@
+#-- copyright
+# OpenProject is a project management system.
+# Copyright (C) 2012-2014 the OpenProject Foundation (OPF)
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See doc/COPYRIGHT.rdoc for more details.
+#++
+
+require 'spec_helper'
+describe OpenProject::Plugins::ModuleHandler do
+  let(:disabled_modules) { OpenProject::Plugins::ModuleHandler.disable_modules('repository') }
+
+  before { OpenProject::Plugins::ModuleHandler.disable(disabled_modules) }
+
+  describe 'disable module' do
+    it { expect(Redmine::AccessControl.available_project_modules).to_not include(:repository) }
+  end
```

Comment:
This will probably cause other tests to fail.

As :repository is removed by the tests, I expect all subsequent specs wanting to ensure repository functionality to fail. The repository permissions therefore have to be reprovided after the test.

Additionally, the other methods of Redmine::AccessControl (e.g. modules_permissions )should be tested as well.